internal/operator-controller/applier/helm.go

```diff
@@ -79,25 +82,65 @@ func shouldSkipPreflight(ctx context.Context, preflight Preflight, ext *ocv1.Clu
 	return false
 }
 
+// runPreAuthorizationChecks performs pre-authorization checks for a Helm release
+// it renders a client-only release, checks permissions using the PreAuthorizer
+// and returns an error if authorization fails or required permissions are missing
+func (h *Helm) runPreAuthorizationChecks(ctx context.Context, ext *ocv1.ClusterExtension, chart *chart.Chart, values chartutil.Values, post postrender.PostRenderer) error {
+	tmplRel, err := h.renderClientOnlyRelease(ctx, ext, chart, values, post)
+	if err != nil {
+		return fmt.Errorf("failed to get release state using client-only dry-run: %w", err)
+	}
+
+	missingRules, authErr := h.PreAuthorizer.PreAuthorize(ctx, ext, strings.NewReader(tmplRel.Manifest))
+
+	var preAuthErrors []error
+
+	if len(missingRules) > 0 {
+		var missingRuleDescriptions []string
+		for _, policyRules := range missingRules {
+			for _, rule := range policyRules.MissingRules {
+				missingRuleDescriptions = append(missingRuleDescriptions, ruleDescription(policyRules.Namespace, rule))
+			}
+		}
+		slices.Sort(missingRuleDescriptions)
+		preAuthErrors = append(preAuthErrors, fmt.Errorf("service account lacks permission to manage cluster extension:\n  %s", strings.Join(missingRuleDescriptions, "\n  ")))
```

Comment:
not sure, but maybe something like this would be better?

```suggestion
		preAuthErrors = append(preAuthErrors, fmt.Errorf("service account requires the following permissions to manage cluster extension:\n  %s", strings.Join(missingRuleDescriptions, "\n  ")))
```